### ISSUE TITLE
Some programs are not recognize devices because of wrong letter convertion on Turkish lanuguage.

### ISSUE THREAD

**Issue by queeup** [karelzak/util-linux#1302]
In Turkish language there is big letter `I` < `ı` and big letter `İ` < `i`. 

Because of this when I am checking devices with `lsblk`:
_Look at key names: `mountpoInt` supposed to be `mountpoint` etc._
```bash
~ $ lsblk --bytes --all --json --paths --output-all
{
   "blockdevices": [
            {"name": "/dev/sdc", "kname": "/dev/sdc", "maj:mIn": "8:32", "fstype": null, "mountpoInt": null, "label": null, "uuId": null, "parttype": null, "partlabel": null, "partuuId": null, "partflags": null, "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": "DataTraveler 2.0", "serIal": "000AEB91BC8CF9A1A5500DDF", "sIze": "4002910208", "state": "running", "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "disk", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": null, "hctl": "8:0:0:0", "tran": "usb", "subsystems": "block:scsi:usb:pci", "rev": "1.00", "vendor": "Kingston", "zoned": "none",
         "children": [
            {"name": "/dev/sdc1", "kname": "/dev/sdc1", "maj:mIn": "8:33", "fstype": "exfat", "mountpoInt": null, "label": "4GB", "uuId": "69AE-7D3C", "parttype": "0x7", "partlabel": null, "partuuId": "f662eea2-01", "partflags": "0x80", "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": null, "serIal": null, "sIze": "4001366016", "state": null, "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "part", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": "/dev/sdc", "hctl": null, "tran": null, "subsystems": "block:scsi:usb:pci", "rev": null, "vendor": null, "zoned": "none"}
         ]
      },
      {"name": "/dev/sdd", "kname": "/dev/sdd", "maj:mIn": "8:48", "fstype": null, "mountpoInt": null, "label": null, "uuId": null, "parttype": null, "partlabel": null, "partuuId": null, "partflags": null, "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": "DataTraveler G3 ", "serIal": "001372982FFBEBC145FC0290", "sIze": "7803174912", "state": "running", "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "disk", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": null, "hctl": "9:0:0:0", "tran": "usb", "subsystems": "block:scsi:usb:pci", "rev": "PMAP", "vendor": "Kingston", "zoned": "none",
         "children": [
            {"name": "/dev/sdd1", "kname": "/dev/sdd1", "maj:mIn": "8:49", "fstype": "exfat", "mountpoInt": null, "label": null, "uuId": "28B7-94BC", "parttype": "0x7", "partlabel": null, "partuuId": "b20e87e2-01", "partflags": null, "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": null, "serIal": null, "sIze": "7801405440", "state": null, "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "part", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": "/dev/sdd", "hctl": null, "tran": null, "subsystems": "block:scsi:usb:pci", "rev": null, "vendor": null, "zoned": "none"}
         ]
      },
      {"name": "/dev/sde", "kname": "/dev/sde", "maj:mIn": "8:64", "fstype": null, "mountpoInt": null, "label": null, "uuId": null, "parttype": null, "partlabel": null, "partuuId": null, "partflags": null, "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": "WiFi Flash Disk ", "serIal": "4443A1825620F018", "sIze": "8022654976", "state": "running", "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "disk", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": null, "hctl": "10:0:0:0", "tran": "usb", "subsystems": "block:scsi:usb:pci", "rev": "1.00", "vendor": "Xiaomi  ", "zoned": "none",
         "children": [
            {"name": "/dev/sde1", "kname": "/dev/sde1", "maj:mIn": "8:65", "fstype": "exfat", "mountpoInt": null, "label": null, "uuId": "4F37-39EB", "parttype": "0x7", "partlabel": null, "partuuId": "3c76bd79-01", "partflags": null, "ra": "128", "ro": "0", "rm": "1", "hotplug": "1", "model": null, "serIal": null, "sIze": "8021606400", "state": null, "owner": "root", "group": "disk", "mode": "brw-rw----", "alIgnment": "0", "mIn-Io": "512", "opt-Io": "0", "phy-sec": "512", "log-sec": "512", "rota": "1", "sched": "deadline", "rq-sIze": "128", "type": "part", "dIsc-aln": "0", "dIsc-gran": "0", "dIsc-max": "0", "dIsc-zero": "0", "wsame": "0", "wwn": null, "rand": "1", "pkname": "/dev/sde", "hctl": null, "tran": null, "subsystems": "block:scsi:usb:pci", "rev": null, "vendor": null, "zoned": "none"}
         ]
      }
   ]
}
```

#### More info:
```bash
~ $ blkid -v
blkid from util-linux 2.34  (libblkid 2.34.0, 14-Jun-2019)

~ $ cat /etc/os-release
NAME="elementary OS"
VERSION="6 Early Access"
ID=elementary
ID_LIKE=ubuntu
PRETTY_NAME="elementary OS 6 Early Access"
VERSION_ID="6"
VERSION_CODENAME=odin
UBUNTU_CODENAME=focal

~ $ locale
LANG=tr_TR.UTF-8
LANGUAGE=tr
LC_CTYPE="tr_TR.UTF-8"
LC_NUMERIC=tr_TR.UTF-8
LC_TIME=tr_TR.UTF-8
LC_COLLATE="tr_TR.UTF-8"
LC_MONETARY=tr_TR.UTF-8
LC_MESSAGES="tr_TR.UTF-8"
LC_PAPER=tr_TR.UTF-8
LC_NAME=tr_TR.UTF-8
LC_ADDRESS=tr_TR.UTF-8
LC_TELEPHONE=tr_TR.UTF-8
LC_MEASUREMENT=tr_TR.UTF-8
LC_IDENTIFICATION=tr_TR.UTF-8
LC_ALL=
```

#### Example issue for this problem:
https://github.com/balena-io/etcher/issues/2731#

**Comment by karelzak** [comment]
Fixed.

Frankly, I do not understand how you can use tr_TR.UTF-8. It seems like a bug that toupper() and tolower() functions are affected by locales when it translating very basic ASCII chars (<=127). It has to be a pretty common problem (for the Turkish language) which is unexpected by developers.

**Comment by lurch** [comment]
Trying to "lower-case" or "upper-case" a (locale-sensitive) string by calling tolower() or toupper() on each character in turn is arguably the wrong approach anyway? I think the proper "localisation friendly" approach would be to have language-independent keys, and a lookup-map where the keys are then translated into the localised strings (which get displayed to the user).

But the key-names being used here aren't locale-sensitive in any case (i.e. it displays "mountpoint" regardless of whether the locale is set to en_US.utf8 or tr_TR.UTF-8) :shrug:   (obviously you wouldn't expect your JSON key-names to suddenly change when running with a different locale setting!)

TL;DR  Localisation is hard to do properly, as lots of languages have weird rules that you don't expect :joy:

**Comment by karelzak** [comment]
The keys are not based on translated strings at all, the problem was that code used locale-sensitive functions ...

**Comment by lurch** [comment]
Indeed, that's what I meant with "But the key-names being used here aren't locale-sensitive in any case" :slightly_smiling_face: 
(so thinking about it some more, I'm not sure why your fix special-cases only chars <= 127 rather than _always_ using `c_toupper` / `c_tolower` ? Perhaps a more "localisation friendly" fix would be for `fputs_quoted_case_json` to gain an extra parameter which tells it whether the strings it's dealing with should be treated as locale-sensitive or not?)

Of course these are only my opinions; this is your code so it's entirely up to you how you fix things :+1:  (and my locale is `en_GB.UTF-8` so I'm not directly affected by these problems anyway!)
